### PR TITLE
Zofiawarzycha dev

### PR DESCRIPTION
**Pull request by zofiawarzycha**

### CODE REVIEW

**Review by Kostya-Iva — APPROVED** [review]
- Wydaję mi się że parametr ile dany całość zajmowała czasu może być ważny, a go brakuje

**Plusy:**
- Duża ilość testów i analizy 
- Wizualizacje :) (nie pomyślałem nad użyciem "Confusion Matrix")
- Całość technicznie, wydaję się być dobrze zrobiona

**Comment by farqlia** [comment]
Podejście do badań jest poprawne i uporządkowane, moje spostrzeżenia:

- mega plus za wydzielenie funkcji do trenowania
- do badania dropout'u można było to dodać jako argument do sieci, zamiast powielać kod dla CNN i hardkodować wartość
- accuracy dla dropout widzę (po kolei): 99.08%, 99.16%, 99.32%, więc w sumie dla wartości 0.2 byłby najlepszy (Albo patrzyłaś na zbiór treningowy, w co wątpię po dalszym czytaniu pracy, albo po prostu przy drugim uruchomieniu były inne wyniki). IMO takie różnice już nie mieszczące się w zakresie 1 pkt. proc. to statystycznie może być brak różnicy (tj. w wielu uruchomieniach modelu, od różnych losowych wag początkowych, średnie dla każdej wartości dropout by wyszły te same)
- mogłaś jaśniej dla punktów 2) i 4) wyjaśnić czym się różnią architektury
- fajnie by było przy takiej ilości eksperymentów zrobić jakiś barplot, żeby porównania między modelami dla metryki i loss (powiedzmy, że oddzielnie dla każdego podzbioru) były czytelniejsze
- dla prostoty kodu, każdy blok postaci: conv ,bn relu i pool można opakować w [nn.Sequential](https://docs.pytorch.org/docs/stable/generated/torch.nn.Sequential.html), wtedy unikniesz tych zagnieżdżonych wywołań w funkcji forward
- macierz pomyłek można też było dodać w niektórych badaniach wyżej, dla zbioru treningowego i walidacyjnego
- brakuje mi trochę finalnych wniosków i też w trakcie badań nie było zapisanych obserwacji co do krzywych uczenia; jest to jednak praca obszerna w eksperymenty, więc można wybaczyć

**Review by Tomoo3 — COMMENTED** [review]
Obszerne rozwiązanie, plus za dużą ilość testów dla różnych wariantów, fajnie jakby było widać ile czasu zajmował trening sieci, a tak ogólnie wszystko dobrze

**Review by fidok15 — APPROVED** [review]
- podwójne zadładowanie zbioru raz pytorchem, raz tensorflowem
- spoko prosty baseline na + 
- dokładne i klarowne dostrajanie modelu na +
- raczej nie ma przeuczenia, ale tutaj jak na treningowym dochodzimy do 99.15 a testowy ma dropa do 98.99 może jakiaś unlucky pulla 
- takich końcowych wniosków i przemyśleń mi brakuje
